contracts/SportMarkets/LiquidityPool/SportAMMLiquidityPool.sol

```diff
@@ -0,0 +1,678 @@
+    /// @param user address to check
+    /// @return neededStaked how much the user needs to have staked to withdraw
+    function getNeededStakedThalesToWithdrawForUser(address user) external view returns (uint neededStaked) {
+        uint nextRound = round + 1;
+        neededStaked = ((balancesPerRound[round][user] + balancesPerRound[nextRound][user]) * ONE) / stakedThalesMultiplier;
+    }
+
+    /// @notice get the pool address for the market
+    /// @param market to check
+    /// @return roundPool the pool address for the market
+    function getMarketPool(address market) external view returns (address roundPool) {
+        roundPool = roundPools[getMarketRound(market)];
+    }
+
+    /// @notice Checks if all conditions are met to close the round
+    /// @return bool
+    function canCloseCurrentRound() public view returns (bool) {
+        if (!started || block.timestamp < getRoundEndTime(round)) {
+            return false;
+        }
+        for (uint i = 0; i < tradingMarketsPerRound[round].length; i++) {
+            IPositionalMarket market = IPositionalMarket(tradingMarketsPerRound[round][i]);
+            if ((!market.resolved())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /// @notice Iterate all markets in the current round and return true if at least one can be exercised
+    function hasMarketsReadyToBeExercised() public view returns (bool) {
+        SportAMMLiquidityPoolRound poolRound = SportAMMLiquidityPoolRound(roundPools[round]);
+        ISportPositionalMarket market;
+        for (uint i = 0; i < tradingMarketsPerRound[round].length; i++) {
+            market = ISportPositionalMarket(tradingMarketsPerRound[round][i]);
+            if (market.resolved()) {
+                (uint homeBalance, uint awayBalance, uint drawBalance) = market.balancesOf(address(poolRound));
+                if (homeBalance > 0 || awayBalance > 0 || drawBalance > 0) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /// @notice Return multiplied PnLs between rounds
+    /// @param roundA Round number from
+    /// @param roundB Round number to
+    /// @return uint
+    function cumulativePnLBetweenRounds(uint roundA, uint roundB) public view returns (uint) {
+        return (cumulativeProfitAndLoss[roundB] * profitAndLossPerRound[roundA]) / cumulativeProfitAndLoss[roundA];
+    }
+
+    /// @notice Return the start time of the passed round
+    /// @param round number
+    /// @return uint the start time of the given round
+    function getRoundStartTime(uint round) public view returns (uint) {
+        return firstRoundStartTime + (round - 1) * roundLength;
+    }
+
+    /// @notice Return the end time of the passed round
+    /// @param round number
+    /// @return uint the end time of the given round
+    function getRoundEndTime(uint round) public view returns (uint) {
```

Comment:
Change to `uint _round` since `round` is a state variable.

Reply:
done thx